WooCommerce/Classes/POS/Presentation/Item Selector/ItemList.swift

```diff
@@ -181,7 +181,9 @@ private struct ItemListRow: View {
             })
         case let .coupon(coupon):
             Button(action: {
-                itemActionHandler.handleTap(item)
+                if !coupon.isExpired {
+                    itemActionHandler.handleTap(item)
+                }
```

Comment:
We could use `.disabled(coupon.isExpired)` here for the button instead, but this removes the tap gesture and animations, which doesn't feel great when they tap on expired coupons. By wrapping the action handler we'll still see the button responding, but won't be added to the cart.